networking.7 : create network quickstart guide

concussious opened the pull request:
Now that the handbook has been moved to ports, I think it's very nice to have a network quickstart guide in-band, in base, in the system manual. If the user uses any of the following terms "man -k {network,networking,wifi,quickstart}" this page will come up, which is I think a very, very common use case for new users.

FreeBSD is the server star, but new users will need to play with it on their laptops before trying to use it in production, right?

Currently, this document explains connecting to a basic Ethernet network, USB tethering, a basic wifi network, scanning for wifi networks, and airplane mode, as well as linking to a few other relevant pages and sections, including the handbook.

grahamperrin commented:
Sorry, ignore the _urndis(4)_ pollution above. A consequence of me attempting to work with a PR in a fork in advance of a simple PR hereabouts.

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+.Sh EXAMPLES
+.Ss Connecting to an ethernet network with DHCP:
+.Dl # dhclient em0
+.Ss Connecting to a wifi network:
+Identify your wifi hardware:
+.Dl % sysctl net.wlan.devices
+Configure your wifi hardware as wlan0 interface:
+.Dl # sysrc wlans_iwn0="wlan0"
+Set that interface to negotiate a dhcp lease with wpa supplicant:
+.Dl # sysrc ifconfig_wlan0="WPA SYNCDHCP"
+Enter the wifi network's details:
+.Dl # echo "network={ssid="myssid" psk="mypsk"} >> wpa_supplicant.conf
+Restart the network interface daemon:
+.Dl # service netif restart
+.Ss Scanning for wifi networks:
```
```suggestion
.Ss Connecting to a Wi-Fi network:
Identify your Wi-Fi hardware:
.Dl % sysctl net.wlan.devices
Configure your Wi-Fi hardware as wlan0 interface:
.Dl # sysrc wlans_iwn0="wlan0"
Set that interface to negotiate a dhcp lease with wpa supplicant:
.Dl # sysrc ifconfig_wlan0="WPA SYNCDHCP"
Enter the details of the Wi-Fi network:
.Dl # echo "network={ssid="myssid" psk="mypsk"} >> wpa_supplicant.conf
Restart the network interface daemon:
.Dl # service netif restart
.Ss Scanning for Wi-Fi networks:
```

Probably more accurate to say _Wi-Fi_.

```text
% rg --count --sort path -e 'Wi-Fi' /usr/doc
/usr/doc/documentation/content/de/books/handbook/advanced-networking/_index.adoc:2
/usr/doc/documentation/content/el/books/handbook/advanced-networking/_index.adoc:3
/usr/doc/documentation/content/en/books/handbook/advanced-networking/_index.adoc:2
/usr/doc/documentation/content/en/books/handbook/advanced-networking/_index.po:2
/usr/doc/documentation/content/es/books/handbook/advanced-networking/_index.adoc:2
/usr/doc/documentation/content/es/books/handbook/advanced-networking/_index.po:6
/usr/doc/documentation/content/fr/books/handbook/advanced-networking/_index.adoc:1
/usr/doc/documentation/content/hu/books/handbook/advanced-networking/_index.adoc:1
/usr/doc/documentation/content/ja/books/handbook/advanced-networking/_index.adoc:1
/usr/doc/documentation/content/mn/books/handbook/advanced-networking/_index.adoc:2
/usr/doc/documentation/content/nl/books/handbook/advanced-networking/_index.adoc:3
/usr/doc/documentation/content/pl/books/handbook/advanced-networking/_index.adoc:3
/usr/doc/documentation/content/pt-br/books/handbook/advanced-networking/_index.adoc:2
/usr/doc/documentation/content/ru/books/handbook/advanced-networking/_index.adoc:1
/usr/doc/documentation/content/zh-cn/books/handbook/advanced-networking/_index.adoc:3
/usr/doc/documentation/content/zh-tw/books/handbook/advanced-networking/_index.adoc:3
/usr/doc/website/content/en/releases/12.4R/errata.adoc:1
/usr/doc/website/content/en/releases/13.2R/errata.adoc:1
/usr/doc/website/content/en/status/report-2017-07-2017-09.html:1
/usr/doc/website/content/en/status/report-2018-01-2018-09.html:1
/usr/doc/website/content/en/status/report-2022-07-2022-09/wtap.adoc:1
/usr/doc/website/content/en/status/report-2023-04-2023-06/batman.adoc:1
/usr/doc/website/data/en/events/events2022.toml:1
/usr/doc/website/static/security/advisories/FreeBSD-EN-22:18.wifi.asc:1
/usr/doc/website/static/security/advisories/FreeBSD-SA-17:07.wpa.asc:4
/usr/doc/website/static/security/advisories/FreeBSD-SA-19:03.wpa.asc:2
/usr/doc/website/static/security/advisories/FreeBSD-SA-22:02.wifi.asc:2
/usr/doc/website/static/security/advisories/FreeBSD-SA-22:07.wifi_meshid.asc:3
/usr/doc/website/static/security/advisories/FreeBSD-SA-23:11.wifi.asc:4
/usr/doc/website/static/security/patches/SA-19:03/wpa-11.patch:44
/usr/doc/website/static/security/patches/SA-19:03/wpa-12.patch:39
% 
```

Avoid contractions. 

<https://docs.freebsd.org/en/books/fdp-primer/book/#writing-style-guidelines>

Thanks

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+.Nm wifi
+.Nd quickstart guide to connecting to a network
+.Sh EXAMPLES
+.Ss Connecting to an ethernet network with DHCP:
```
```suggestion
.Ss Connecting to an Ethernet network with DHCP:
```

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+.Dl % sysctl net.wlan.devices
+Configure your wifi hardware as wlan0 interface:
+.Dl # sysrc wlans_iwn0="wlan0"
+Set that interface to negotiate a dhcp lease with wpa supplicant:
```
```suggestion
Set that interface to negotiate a DHCP lease with
.Xr wpa_supplicant 8 :
```

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+.Xr wpa_supplicant 8
+.Pp
+.Fx
+handbook chapter 42: Advanced Networking
```
```suggestion
The Advanced Networking chapter of the FreeBSD Handbook
```

The number is currently 34 (not 42), and may change.

42 is [reserved for future use](https://en.wikipedia.org/wiki/Phrases_from_The_Hitchhiker%27s_Guide_to_the_Galaxy#The_Answer_to_the_Ultimate_Question_of_Life,_the_Universe,_and_Everything_is_42).

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+Identify your wifi hardware:
+.Dl % sysctl net.wlan.devices
+Configure your wifi hardware as wlan0 interface:
+.Dl # sysrc wlans_iwn0="wlan0"
```
I think, the page would need something to explain the iwn(4) assumption.

concussious replied:
Agree, added to DESCRIPTION.

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+Enter the wifi network's details:
+.Dl # echo "network={ssid="myssid" psk="mypsk"} >> wpa_supplicant.conf
+Restart the network interface daemon:
+.Dl # service netif restart
```
Commonly recommended, however it's not reliable. For example, from <https://lists.freebsd.org/archives/freebsd-current/2023-August/004133.html>: 

>> … so `service routing restart` must be run quickly after `service netif restart` - before we receive dhcp offer. …

Maybe add a BUGS section to the page?

concussious replied:
After the change to mention wpa_passphrase, I felt that mentioning escaping shell special characters is probably necessary for the new users this document is primarily targeted at. Since this is not a bug, I added both of these to a CAVEATS section.

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+Set that interface to negotiate a dhcp lease with wpa supplicant:
+.Dl # sysrc ifconfig_wlan0="WPA SYNCDHCP"
+Enter the wifi network's details:
+.Dl # echo "network={ssid="myssid" psk="mypsk"} >> wpa_supplicant.conf
```
Maybe better to exemplify use of [wpa_passphrase(8)](https://man.freebsd.org/cgi/man.cgi?query=wpa_passphrase&sektion=8&manpath=freebsd-release).

concussious replied:
Agree, included in latest commit

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+.Xr bsdconfig 8 ,
+.Xr dhclient 8 ,
+.Xr ifconfig 8 ,
+.Xr wpa_supplicant 8
```
```suggestion
.Xr wpa_passphrase 8
.Xr wpa_supplicant 8
```

concussious replied:
My latest commit includes both, how do you feel about this?

grahamperrin commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,59 @@
+.Dd August 29, 2023
+.Dt "NETWORKING" 7
+.Os
+.Sh NAME
+.Nm networking ,
+.Nm wifi
+.Nd quickstart guide to connecting to a network
+.Sh EXAMPLES
```
From <https://man.freebsd.org/cgi/man.cgi?query=mdoc&sektion=7&manpath=freebsd-release#MANUAL_STRUCTURE> in mdoc(7): 

> … convention dictates specifying at least the SYNOPSIS and DESCRIPTION sections, although this varies between manual sections. …

concussious replied:
I didn't feel SYNOPSIS was relevant, most pages in section 7 didn't have them. My latest commit adds a description attempting to explain the nomenclature of interfaces.

concussious commented:
> Sorry, ignore the _urndis(4)_ pollution above. A consequence of me attempting to work with a PR in a fork in advance of a simple PR hereabouts.

No, that's beautiful. I completely forgot about USB tethering. It should be mentioned here.

concussious commented:
I think it's less legible, but I rewrote the examples to format accordingly to style.mdoc(5), except without the depreciated Li macro. The guideline laid out there breaks a lot of mdoc conventions. The It macro is stated in mdoc(7) to be for individual commands and explicitly not entire lines. Also, subsections have their own macro, and I think they're appropriate for this use case. I have a copy with all the changes but the earlier formatting. Thoughts?

bsdimp commented on share/man/man7/networking.7:
```diff
@@ -0,0 +1,122 @@
+.\" SPDX-License-Identifier: BSD-2-Clause
+.\"
+.\" Copyright (c) 2023 Alexander Ziaee
+.\"
```
If you want, you can omit the actual text of the license now.

concussious replied:
Thank you, added in latest commit

bsdimp commented:
This commit needs to have a real email, not the noreply github one,

igalic commented:
> This commit needs to have a real email, not the noreply github one,

@concussious this is still not addressed

concussious commented:
> > This commit needs to have a real email, not the noreply github one,
> 
> @concussious this is still not addressed

I'm sorry, I thought that I did, and I don't understand.

igalic commented:
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

you need to set the commit email in your git cli, then go back rewrite the author: https://stackoverflow.com/questions/750172/how-do-i-change-the-author-and-committer-name-email-for-multiple-commits#1320317

concussious commented:
Should I squash all these commits down to one like I was instructed to in the other thread? If so, will that negatively impact the commits coauthored with grahamperrin?

igalic commented:
I asked @grahamperrin to chime in on whether he minds getting squashed. meanwhile, you can credit yourself as the author of this man page, if you so desire, on the bottom in an `.Sh AUTHORS` section.

concussious commented:
In such a case should we add him to the copyright at the top?

igalic commented:
> In such a case should we add him to the copyright at the top?

if you want, sure. your call

bsdimp commented:
@bsdbcr Can you review this in more detail. It seems OK to me, but I'm not sure it belongs here. Thanks!

bsdbcr commented:
> @bsdbcr Can you review this in more detail. It seems OK to me, but I'm not sure it belongs here. Thanks!

I've reviewed the changes. That looks solid and would be a good addition to our docs. Nothing to change, add, or complain about.

concussious commented:
I'm also concerned about I don't want to mislead anyone. The first step of connecting to a network is configuring a firewall, unless the entire machine is already behind a firewall, which I think is kind of an advanced use case.

concussious commented:
I just discovered intro(7) lists all the pages in section seven. Therefore, I have added this and everything else my machine has in section seven to intro(7).

While there, I also updated the history section to the same format used in the new hier(7).

concussious commented:
> I'm not sure it belongs here.

Is it appropriate for `netintro.4`?

bsdimp commented:
OK. This isn't getting any better, and hans't for a while. It looks good as is. I'll leave the intro issue for others.
